### PR TITLE
sort mariadb gtid set

### PR DESCRIPTION
**Pull request by zeminzhou**
当 mariadb gtid中包含多个 server id时，由于 Sets是一个map结构，输出 set的顺序随机，导致最终输出的 gtid字符串内容不同。本 PR，在输出前对内容进行排序后输出，保证输出结果的一致。

### CODE REVIEW

**Comment by atercattus** [comment]
Hello, @zeminzhou .

Unfortunately, this repo is abandoned. I made [PR about this situation](https://github.com/siddontang/go-mysql/blob/4f95e0c261b51f2a1962b00a1c0dedc96622af24/README.md#this-repo-is-abandoned
), but could not merge it :)

I'll be glad to see your PR in [my fork](https://github.com/atercattus/go-mysql), which I continue to support and develop.

**Comment by lance6716** [comment]
Hi @zeminzhou, please add a unit test about this behaviour

**Comment by zeminzhou** [comment]
> Hi @zeminzhou, please add a unit test about this behaviour

I'm sorry I just saw this email notification.  Glad to read your comments, I will add a unit test for this pr tomorrow.

File: mysql/mariadb_gtid.go [hunk 1]
```diff
@@ -157,7 +158,26 @@ func (s *MariadbGTIDSet) Update(GTIDStr string) error {
...
+	}
+
+	var buf bytes.Buffer
+	sets := make([]string, 0, len(s.Sets))
+	for _, set := range s.Sets {
+		sets = append(sets, set.String())
+	}
+	sort.Strings(sets)
+
+	sep := ""
```
Comment (lance6716) [inline mysql/mariadb_gtid.go:174]: we could use `strings.Join`

File: mysql/mariadb_gtid_test.go [hunk 2]
```diff
@@ -232,3 +232,12 @@ func (t *mariaDBTestSuite) TestMariaDBGTIDSetClone(c *check.C) {
 		c.Assert(gtidSet.Clone(), check.DeepEquals, gtidSet)
 	}
 }
+
+func (t *mariaDBTestSuite) TestMariaDBGTIDSetSortedString(c *check.C) {
+	gtidSetStr := "2-2-2,1-1-1,3-2-1"
```
Comment (lance6716) [inline mysql/mariadb_gtid_test.go:237]: please also test GTIDSet size of zero and one
